### PR TITLE
Adds routing from Auth -> Onboarding -> Home

### LINKED ISSUE #519: [Onboarding] Present on-boarding experience/routes when first opening app
### User Story

As a user I should be presented the `Auth` route then the `Onboarding` route every time I log in.

### Description

Everytime a user isn't logged in, we should force the `Auth` route. After login, we should then direct the user to the `Onboarding` route. After `Onboarding`, we can take the user to the home screen.

Since we persist your login (we currently reset this each build but will be changing that) you should only have to login ~1 time per device. Every time you login on a new device, you need to be presented with the onboarding experience again to enable notifications and location services. For MVP, we are comfortable just re-showing onboarding every time you login. Again, we currently reset your auth state with every build, and we will change that to not function that way in a separate task.

### Acceptance Criteria

* [ ] It should show the `Auth` route if you're not logged in
* [ ] After logging in, push/switch to the `Onboard` route
* [ ] After onboarding, go back to the home tab

### Possible Implementation & Open Questions

* The yet-to-be-merge [`ui auth` package](https://github.com/ApollosProject/apollos-prototype/pull/591) adds an `afterAuth` style callback to make some of this easier, but I think we can go ahead and wire this up with the current `Auth` route in `apolloschurchapp` until that's merged
* We probably want to maintain separate `Auth` and `Onboarding` routes on the root navigator

### Other Resources

* Basecamp: https://3.basecamp.com/3926363/buckets/6947926/todolists/1506896432

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Adds routing from Auth -> Onboarding -> Home

### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

1. Signout of the app.
2. Refresh the app, and login
3. Navigate through onboarding
4. You should end up on the home screen.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes #519 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [x] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-auth/src/index.js [hunk 1]
```diff
@@ -71,7 +71,6 @@ const CancelButton = styled(({ theme }) => ({
 class Auth extends PureComponent {
   static navigationOptions = {
     header: null,
-    gesturesEnabled: false,
```
Comment (burkeshartsis) [inline packages/apollos-ui-auth/src/index.js]: 🙌I love how this is no longer needed!

File: packages/apolloschurchapp/src/auth/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Auth from '@apollosproject/ui-auth';
+
+const AuthWithOnboarding = (props) => (
```
Comment (burkeshartsis) [inline packages/apolloschurchapp/src/auth/index.js]: 🏗

File: packages/apolloschurchapp/src/index.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { BackgroundView, withTheme } from '@apollosproject/ui-kit';
 import Passes from '@apollosproject/ui-passes';
 
 import MediaPlayer from 'apolloschurchapp/src/ui/MediaPlayer';
-import Auth, { ProtectedRoute } from '@apollosproject/ui-auth';
+import { ProtectedRoute } from '@apollosproject/ui-auth';
```
Comment (burkeshartsis) [inline packages/apolloschurchapp/src/index.js]:
```suggestion
import { ProtectedRoute } from '@apollosproject/ui-auth';

```

File: packages/apolloschurchapp/src/index.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { BackgroundView, withTheme } from '@apollosproject/ui-kit';
 import Passes from '@apollosproject/ui-passes';
```
Comment (burkeshartsis) [inline packages/apolloschurchapp/src/index.js]:
```suggestion
```

File: packages/apolloschurchapp/src/ui/Onboarding/index.js [hunk 5]
```diff
@@ -23,34 +20,14 @@ const ThemedSwiper = withTheme(({ theme }) => ({
...
-  backgroundColor: bgcolor,
-}))(FlexedView);
-
 class Onboarding extends Component {
   constructor() {
     super();
 
     this.swiper = null;
-    this.state = { pagination: true };
+    this.state = { pagination: true, swipingEnabled: true };
```
Comment (burkeshartsis) [inline packages/apolloschurchapp/src/ui/Onboarding/index.js]: I don't think `pagination` is used any more. If so it should be removed.

Additionally, what are we doing with state anyways? What is `swipingEnabled` in state for?

Comment (vinnyjth) [reply packages/apolloschurchapp/src/ui/Onboarding/index.js]: That's old and left over, I will fix :)

**Pull request merged**